lib/mumukit/auth/grant.rb

```diff
@@ -38,39 +41,32 @@ def to_s
     end
   end
 
-  class SingleGrant < Grant
-    def initialize(slug)
-      @slug = slug
-    end
-
-    def allows?(slug)
-      @slug == slug
+  class FirstPartGrant < Grant
+    def initialize(first)
+      @first = first
     end
 
-    def access?(organization)
-      @slug.split('/')[0] == organization
+    def allows?(resource_slug)
+      resource_slug.to_mumukit_slug.match_first @first
     end
 
     def to_s
-      @slug
+      "#{@first}/*"
     end
   end
 
-  class OrgGrant < Grant
-    def initialize(org)
-      @org = org
-    end
-
-    def allows?(slug)
-      /^#{@org}\/.*/.matches? slug
+  class SingleGrant < Grant
+    def initialize(slug)
+      @slug = slug
     end
 
-    def access?(organization)
-      @org == organization
+    def allows?(resource_slug)
+      resource_slug = resource_slug.to_mumukit_slug
+      resource_slug.match_first(@slug.first) && resource_slug.match_second(@slug.second)
```

Comment:
But I need something composable here. The way `AllGrant` and `FirstPartGrant` compare resources is different - it depends on the grant, not the slug.